azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -84,6 +84,28 @@
     this.dependencyTransferManager = dependencyTransferManager;
   }
 
+  /**
+   * Constructor
+   * This constructor is used for Containerized execution which does not need projectCache
+   *
+   * @param projectStorageManager projectStorageManager
+   * @param dependencyTransferManager dependencyStorageManager
+   */
+  public FlowPreparer(final ProjectStorageManager projectStorageManager,
+               final DependencyTransferManager dependencyTransferManager) {
+    Preconditions.checkNotNull(projectStorageManager);
+    Preconditions.checkNotNull(dependencyTransferManager);
+
+    this.projectStorageManager = projectStorageManager;
+    this.dependencyTransferManager = dependencyTransferManager;
+
+    // Not needed.
+    this.executionsDir = null;
```

Comment:
Why is executionDir null?

Reply:
executionDir is the executions directory in executor server, there is none in containerized flow.

Reply:
Where will the job and flow logs will go? In flow container directory?